AdobeStockImage/Test/Integration/Model/SaveImageTest.php

```diff
@@ -0,0 +1,219 @@
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Test\Integration\Model;
+
+use AdobeStock\Api\Models\StockFile;
+use Magento\AdobeStockAssetApi\Api\AssetRepositoryInterface;
+use Magento\AdobeStockClient\Model\StockFileToDocument;
+use Magento\AdobeStockImage\Model\SaveImageFile;
+use Magento\AdobeStockImage\Model\Storage\Save;
+use Magento\AdobeStockImageApi\Api\SaveImageInterface;
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\Search\Document;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\IntegrationException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Driver\Https;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test client for communication to Adobe Stock API.
+ */
+class SaveImageTest extends TestCase
+{
+    const URL_FIELD = 'thumbnail_240_url';
+
+    /**
+     * @var SaveImageInterface
+     */
+    private $saveImage;
+
+    /**
+     * @var string
+     */
+    private $saveDestination = 'catalog/category/tmp.png';
+
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var Filesystem
+     */
+    private $fileSystem;
+
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $criteriaBuilder;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->driver = Bootstrap::getObjectManager()->get(DriverInterface::class);
+        $this->fileSystem = Bootstrap::getObjectManager()->get(Filesystem::class);
+        $this->assetRepository = Bootstrap::getObjectManager()->get(AssetRepositoryInterface::class);
+        $this->criteriaBuilder = Bootstrap::getObjectManager()->get(SearchCriteriaBuilder::class);
+
+        $this->deleteImage();
+        $https = $this->createMock(Https::class);
+        $https->expects($this->once())
+            ->method('fileGetContents')
+            ->willReturnCallback(function ($filePath) {
+                return file_get_contents($filePath);
+            });
+        $storageSave = Bootstrap::getObjectManager()->create(Save::class, ['driver' => $https]);
+        $saveImageFile = Bootstrap::getObjectManager()->create(SaveImageFile::class, ['storageSave' => $storageSave]);
+        $this->saveImage = Bootstrap::getObjectManager()->create(
+            SaveImageInterface::class,
+            ['saveImageFile' => $saveImageFile]
+        );
+    }
+
+    /**
+     * @inheridoc
+     */
+    protected function tearDown(): void
+    {
+        $this->deleteImage();
+        parent::tearDown();
+    }
+
+    /**
+     * Test with image.
+     *
+     * @return void
+     */
+    public function testSave(): void
```

Comment:
It would be good to provide these values from the data provider
```suggestion
    public function testSave(array $documentData, string $sourcePath, string $destinationPath): void
```